snowplow_tracker/emitters.py

```diff
@@ -314,7 +339,12 @@ def set_flush_timer(self, timeout: float, flush_now: bool = False) -> None:
 
         # Repeatable create new timer
         if flush_now:
-            self.flush()
```

Comment:
I just realized that we have abused the `set_flush_timer()` function for a purpose that it was not intended to be used... It actually won't work for our purpose because the function starts a never-ending timer that keeps repeating in an interval. In our case, we only want to fire the timer once after the retry delay. I think the original purpose of the function was to let the user flush the events periodically in an interval.

So unfortunately, we need to change this. We should make a second timer `self.retry_timer` and another function `_set_retry_timer(self, timeout: float, flush_now: bool = False)` (the leading underscore to indicate it's not public API for users) with the same content as this function except that it should return from the `if flush_now:` block. We should also undo the changes in this function.